[12.0][IMP] web_m2x_options: Added limit for o2m field entries + efficiency

CarlosRoca13 opened the pull request:
cc @Tecnativa TT26667

Added the system parameter "web_m2x_options.field_limit_entries" to set the limit of entries that can be displayed on One2many fields.

Improved the eficiency of code: With this improvement we achieve to just do an rpc call when the js file is called instead of do the call for each field.

Here you have an example:
![1](https://user-images.githubusercontent.com/35952655/98531348-80faa400-2280-11eb-94b9-95b7de2d4f7e.gif)

Please @chienandalu @ernestotejeda can you review this

CarlosRoca13 committed:
[IMP] web_m2x_options: Efficiency
Improved the efficiency of code: With this improvement we achieve to just do an rpc call when the js file is called instead of do the call for each field.

pedrobaeza commented on web_m2x_options/README.rst:
```diff
@@ -115,13 +115,18 @@ If you disable one option, you can enable it for particular field by setting "cr
 
   Whether the field should always show "Search more..." entry or not.
 
+``web_m2x_options.field_limit_entries`` *int* (Default: odoo default value is ``40``)
```
There shouldn't be a default value, as if not defined, the limit is set by Odoo itself in their core, so IMO you shouldn't change that default value if not defined.

pedrobaeza reviewed (changes requested):
The improvement for reading only one time should be split in a different commit from the new feature for limiting the number of entries in the one2many.

CarlosRoca13 commented:
@pedrobaeza Changes done

pedrobaeza commented on web_m2x_options/readme/USAGE.rst:
```diff
@@ -69,13 +69,18 @@ If you disable one option, you can enable it for particular field by setting "cr
 
   Whether the field should always show "Search more..." entry or not.
 
+``web_m2x_options.field_limit_entries`` *int* (Default: odoo default value is ``40``)
```
This change is still in the wrong commit

CarlosRoca13 committed:
[IMP] web_m2x_options: Added limit for o2m field entries
Added the system parameter "web_m2x_options.field_limit_entries" to set the limit of entries that can be displayed.

CarlosRoca13 commented:
Sorry, changed yet @pedrobaeza

joao-p-marques reviewed (approved):
LGTM :+1:

pedrobaeza commented:
/ocabot merge major
